src/test/java/com/gooddata/featureflag/FeatureFlagServiceAT.java

```diff
@@ -67,24 +67,40 @@ public void updateProjectFeatureFlag() throws Exception {
         // disable (update) feature flag
         featureFlag.setEnabled(false);
         final ProjectFeatureFlag disabledFlag = gd.getFeatureFlagService().updateProjectFeatureFlag(featureFlag);
-        // can not check value because of eventual consistency of this API
-        // checkProjectFeatureFlag(disabledFlag, false);
+        checkProjectFeatureFlag(disabledFlag, PROJECT_FEATURE_FLAG, false);
 
         // enable again
         featureFlag.setEnabled(true);
         final ProjectFeatureFlag enabledFlag = gd.getFeatureFlagService().updateProjectFeatureFlag(featureFlag);
-        // can not check value because of eventual consistency of this API
-        // checkProjectFeatureFlag(enabledFlag, true);
+        checkProjectFeatureFlag(enabledFlag, PROJECT_FEATURE_FLAG , true);
     }
 
-    @Test(groups = "featureFlag", dependsOnMethods = "createProjectFeatureFlag")
+    @Test(groups = "featureFlag", dependsOnMethods = "createProjectFeatureFlag", expectedExceptions=GoodDataException.class)
     public void deleteProjectFeatureFlag() throws Exception {
         final ProjectFeatureFlag featureFlag =
                 gd.getFeatureFlagService().createProjectFeatureFlag(project,
                         new ProjectFeatureFlag("settingName1"));
 
         gd.getFeatureFlagService().deleteProjectFeatureFlag(featureFlag);
-        // intentionally don't check the flag is really deleted because of eventual consistency of this API
+        gd.getFeatureFlagService().getProjectFeatureFlag(project, "settingName1");
```

Comment:
I would just replace `expectedExceptions` with `try-catch` surrounding this last call - to prevent successful test in case when e.g. `deleteProjectFeatureFlag` throws exception ;)